src/renderers/dom/fiber/ReactDOMFiber.js

```diff
@@ -347,34 +354,49 @@ var DOMRenderer = ReactFiberReconciler({
     parentInstance: Instance | Container,
     child: Instance | TextInstance,
   ): void {
-    parentInstance.appendChild(child);
+    if (parentInstance.nodeType === COMMENT_NODE) {
```

Comment:
Hm. Maybe it's not so bad since markup generators would likely use the `appendInitialChild` path instead and not this one.

Still, seems cleaner if the `Container` had its own helpers so that hosts don't need to have to build pattern matching into their data structures when we already have the `HostRoot` tag for that.